lib/ipmi_hpmfwupg.c

```diff
@@ -1572,9 +1572,11 @@ HpmfwupgGetComponentProperties(struct ipmi_intf *intf,
 				val2str(rsp->ccode, completion_code_vals));
 		return HPMFWUPG_ERROR;
 	}
+
+	memcpy(&pCtx->resp, rsp->data, rsp->data_len);
+	((uint8_t*)&pCtx->resp)[rsp->data_len] = '\0';
```

Comment:
Generally `rsp->data` is not a string, you can't safely terminate it like this.
Instead you need a `memset(&pCtx->resp, 0, sizeof(pCtx->resp))` before `memcpy()`.

Also this `memcpy(..., rsp->data_len)` is a security vulnerability. You should never copy blindly any size that you got from outer world. Please use `__min(rsp->data_len, sizeof(pCtx->resp))`.

Reply:
Also, maybe this `memset()` should go into `HpmfwupgSendCmd()` instead? Or maybe even deeper, somewhere into the interfaces `sendrecv()` calls. I don't think any of the `sendrecv()` calls should return any garbage in the `rsp`.

Please investigate this further.

Reply:
Well, I agree consolidating memcpy() like this change has a potential security issue. I'll revert the changes of memcpy() and add the __min() to each memcpy() function call.

Ideally, memset() should be in sendrecv(). I agree that sendrecv() calls should  not return any garbage. However, unfortunately, as you can see my changes for the src/plugins/lanplus.c, some of interface drivers do return garbage. I mentioned them in the Issue 289. Since I don't plan to change and test all these drivers, add a memset() here will help when other interface drivers are used.

Since I've been steered away to a totally different area, I'll be very slow on this.